packages/x-charts-pro/src/internals/plugins/useChartProZoom/gestureHooks/usePanOnDrag.ts

```diff
@@ -23,21 +26,46 @@ export const usePanOnDrag = (
   const drawingArea = useSelector(store, selectorChartDrawingArea);
   const optionsLookup = useSelector(store, selectorChartZoomOptionsLookup);
   const startRef = React.useRef<readonly ZoomData[]>(null);
+  const pressedKeysRef = React.useRef<Set<string>>(new Set());
+  const config = useSelector(store, selectorPanConfig, ['onDrag' as const]);
 
   // Add event for chart panning
   const isPanEnabled = React.useMemo(
-    () => Object.values(optionsLookup).some((v) => v.panning) || false,
-    [optionsLookup],
+    () => (Object.values(optionsLookup).some((v) => v.panning) && config) || false,
+    [optionsLookup, config],
   );
 
+  React.useEffect(() => {
+    const pressedKeysSet = pressedKeysRef.current;
+    if (!isPanEnabled || !config) {
+      return () => {};
+    }
+
+    const handleKeyDown = (event: KeyboardEvent) => pressedKeysRef.current.add(event.key);
+    const handleKeyUp = (event: KeyboardEvent) => pressedKeysRef.current.delete(event.key);
+    window.addEventListener('keydown', handleKeyDown);
+    window.addEventListener('keyup', handleKeyUp);
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+      window.removeEventListener('keyup', handleKeyUp);
+      pressedKeysSet.clear();
+    };
+  }, [isPanEnabled, config]);
```

Comment:
unsure if I should move the keys filtering behaviour into the gestures lib

Reply:
If I was the maintainer of the gestures library, it's a feature request I would consider legitimate